### ISSUE TITLE
Website crashing when you refresh results or details page

### ISSUE THREAD

**Issue by georgiamshaw** [fac-17/Generation-Change#92]

**Comment by gminova** [comment]
![1refresh404-details-page](https://user-images.githubusercontent.com/42014388/66908223-9dd39200-f002-11e9-93d6-48b763af81c3.png)
![1refresh404-results-page](https://user-images.githubusercontent.com/42014388/66908225-9e6c2880-f002-11e9-8d28-98b94df48bc6.png)
However in browesr forward/back buttons works correctly, and all airtable data is still reachable at endpoint: https://generation-change.netlify.com/.netlify/functions/getdata

**Comment by georgiamshaw** [comment]
What we think is happening is the state gets lost. 

Solutions we have been thinking of trying are storing our data in local storage or looking into using the useContext hook. 

Any suggestions welcome!

**Comment by maxgerber** [comment]
You're right — as React is a front-end framework your state will not persist between sessions unless you set something up to capture it.

AFAIK useContext will give you the same problem. I'd recommend using the sessionStorage API, which works much like localStorage. However, sessionStorage will only persist so long as your browser is open (including when you refresh), and will delete every time you close your browser.

https://www.robinwieruch.de/local-storage-react

**Comment by samjam48** [comment]
we could just use more of the router capabilities instead of relying on state and session storage so when you reload the url it works it our from the url.
e.g. {site-name}/details/{project_id}

But if we are keeping all the airtable data in our state as well then it would mean we have to do more specific calls to airtable on every different page or still use session storage for the airtable data...
